docs/sphinx/user_guide/run.rst

```diff
@@ -138,3 +138,111 @@ was not possible for them to co-exist in the same executable as CUDA
+        tt = gf.graph.roots[0].traverse(order="pre")
+        for nn in tt:
+            # test if leaf node
+            if not nn.children:
+                for mindex in range(0, 4):
+                    values[findex, mindex] = gf.dataframe.loc[nn, metric[mindex]]
+        findex += 1
+    for mindex in range(0,len(metric)):
+        md[metric[mindex]] = np.average(values[:,mindex])
+    return values,md  
+
+
+> Caveat for the code snippet above. If you run -atsc topdown.toplevel,profile.mpi  then the MPI Barrier routine will be nested at the leaf node vs expected Kernel.Tuning
+
+> Other caveats:  raw counter values are often noisy and require a lot of accommodation to collect accurate data; these include turning off Hyperthreading, turning off Prefetch as is done in Intel's Memory Latency Checker (requires Root), adding LFENCE instruction to serialize and bracket code under test, disable preemption and hard interrupts. See Andreas Abel's dissertation "Automatic Generation of Models of Microarchitectures" for more info on this and for a comprehensive look at the nanobench machinery.
+
+`Yasin's Paper <https://www.researchgate.net/publication/269302126_A_Top-Down_method_for_performance_analysis_and_counters_architecture>`_
+
+`Vtune-cookbook topdown method <https://www.intel.com/content/www/us/en/develop/documentation/vtune-cookbook/top/methodologies/top-down-microarchitecture-analysis-method.html>`_
+
+`Automatic Generation of Models of Microarchitectures <https://uops.info/dissertation.pdf>`_
+
+Generating trace events (time-series) for viewing in chrome://tracing or Perfetto
+---------------------------------------------------------------------------------
+
+`Perfetto <https://ui.perfetto.dev/>`_
+
+Use Caliper's event trace service to collect timestamp info, where kernel timing can be viewed using browser trace profile views
+
+``CALI_CONFIG=event-trace,event.timestamps ./raja-perf.exe -ek PI_ATOMIC INDEXLIST  -sp``
+
+This will produce a separate .cali file with date prefix which looks something like 221108-100718_724_ZKrHC68b77Yd.cali
+
+Then we'll need to convert this .cali file to JSON records, but first we need to make sure Caliper's python reader is available in the PYTHONPATH, 
+
+``export PYTHONPATH=caliper-source-dir/python/caliper-reader``
+
+then run cali2traceevent.py. Example:
+
+``python3 ~/workspace/Caliper/python/cali2traceevent.py 221108-102406_956_9WkZo6xvetnu.cali RAJAPerf.trace.json``
+
+You can then load the resulting JSON file either in Chrome by going to chrome://tracing or in Perfetto.
+
+For CUDA, assuming you built Caliper with CUDA support, you can collect and combine trace information for memcpy, kernel launch, synchronization, and kernels; with example:
+
+``CALI_CONFIG="event-trace(event.timestamps,trace.cuda=true,cuda.activities)" ./raja-perf.exe -v RAJA_CUDA Base_CUDA -k Algorithm_REDUCE_SUM -sp``
+
+However, when you run cali2traceevent.py you need to add --sort option before the filenames.
```

Comment:
What is the sort option for?

Reply:
cali2traceevent needs to sort trace.cuda records before processing. I just noticed that the latest version of that code has --sort builtin as default -- probably because I didn't use that switch it out of the box and complained, so David updated the script